src/components/ListItem.jsx

```diff
@@ -1,9 +1,48 @@
 import './ListItem.css';
+import { useEffect, useCallback } from 'react';
+import { updateItem } from '../api/firebase';
+
+const millisecondsIn24hrs = 86400000;
+
+export function ListItem({ listToken, item }) {
+	useEffect(() => {
+		let calcDate = item.dateLastPurchased
+			? item.dateLastPurchased.seconds * 1000
+			: null;
+		const currentDate = new Date();
+		const currentTime = currentDate.getTime();
+		let timer = currentTime - calcDate;
+		if (item.isChecked && timer >= millisecondsIn24hrs) {
+			updateItem(listToken, item, {
+				isChecked: !item.isChecked,
+			});
+		}
+	}, []);
+
+	const handleChange = useCallback(async () => {
+		if (item.isChecked === false) {
+			await updateItem(listToken, item, {
```

Comment:
updateItem function receives two variables but you pass 3 so ultimately I don' thinks this is working as it is supposed to.

Reply:
the responses are passed from item.  item.isChecked, item.totalPurchases, etc

Reply:
Yeah but you want to update them not keep them the same